openedx/features/course_experience/tests/views/test_course_home.py

```diff
@@ -844,6 +848,46 @@ def test_course_goal_updates(self):
         self.assertContains(response, TEST_COURSE_GOAL_UPDATE_FIELD)
         self.assertNotContains(response, TEST_COURSE_GOAL_UPDATE_FIELD_HIDDEN)
 
+    @override_waffle_flag(COURSE_PRE_START_ACCESS_FLAG, active=True)
+    def test_course_messaging_for_public_course(self):
+        """
+        Ensure that the following use cases work as expected for public courses
+
+        1) Anonymous users are not shown a course message asking them to enroll
+        2) Unenrolled users are not shown a course message asking them to enroll
+        3) Unenrolled users are shown a course message asking them to enroll for courses that allow
+           self-enrollment
+        """
+        TEST_COURSE_ANONYMOUS_MESSAGE = 'You must be enrolled in the course to see course content.'
+        enable_unenrolled_access = True
+
+        with mock.patch('xmodule.course_module.CourseDescriptor.course_visibility', COURSE_VISIBILITY_PUBLIC):
+            with override_waffle_flag(COURSE_ENABLE_UNENROLLED_ACCESS_FLAG, enable_unenrolled_access):
+                # Anonymous users are not shown a course message asking them to enroll
+                url = course_home_url(self.course)
+                response = self.client.get(url)
+                self.assertNotContains(response, TEST_COURSE_ANONYMOUS_MESSAGE)
+
+                # Unenrolled users are not shown a course message asking them to enroll for courses
+                # that do not allow self-enrollment
+                user = self.create_user_for_course(self.course, CourseUserType.UNENROLLED)
+                # Closed course
+                now = datetime.now().replace(tzinfo=pytz.UTC)
+                start = now + timedelta(days=1)
+                end = now + timedelta(days=2)
+                course = CourseFactory(enrollment_start=start, enrollment_end=end)
+                url = course_home_url(course)
+                response = self.client.get(url)
+                self.assertNotContains(response, TEST_COURSE_HOME_MESSAGE_UNENROLLED)
+                self.assertNotContains(response, TEST_COURSE_ANONYMOUS_MESSAGE)
+
+                # Unenrolled users are shown a course message asking them to enroll for courses that allow
+                # self-enrollment
+                user = self.create_user_for_course(self.course, CourseUserType.UNENROLLED)
+                url = course_home_url(self.course)
+                response = self.client.get(url)
+                self.assertContains(response, TEST_COURSE_HOME_MESSAGE_UNENROLLED)
+
```

Comment:
This can be combined with existing tests above in line : https://github.com/edx/edx-platform/blob/master/openedx/features/course_experience/tests/views/test_course_home.py#L315

Reply:
I was looking into this now and I'm not sure about the changes to be made to the test you referred me. Do I have to make changes in the ddt input for the function? I tried to keep my tests separate, but that function seems to test many different features.